### DISCUSSION TITLE
There is no competition between javascript and Blazor both are friends

### DISCUSSION THREAD

**Discussion by jatinshah404** [dotnet/aspnetcore#58378] [Ideas]
.NET Blazor is a really nice technology. You know very well. Still it is not becoming popular like javascript frameworks like Next.JS. My company has a ASP.NET webforms project. My company decided to upgrade that to new technologies. I suggested to use Blazor for that but after a lot of debate company choose to go with Next.js.  

There are key reason behind that.
1. Management feels that it is fast because of smooth execution of javascript.
2. Component libraries like shad cn UI. 
3. DashTail - Tailwind, React Next Admin Dashboard Template

To overcome this situation, you need 

1.	People are feeling that it is competition between c# and javascript. we need to change that narrative and convince that Blazor supports javascript like ASP.NET was also supporting.  It is only interactivity that is done using c# programming language. still animation like rendering charts etc can be done using css and javascript. 
2.	Make more Admin dashboard templates available through third party vendors. There are many open source libraries and javascript libraries supported in blazor. We need to use them same way  people are doing for next js.  We need to combine all various opensource components from multiple vendors and create one Admin template and start selling at almost 0 cost.
3.	Microsoft has FluentUI library that is good but we need to also encourage more open source libraries like Radzen blazor, Sysinfocus simple/ui and make it freely available for all users.
This way Blazor technology will be more. User base will increase and again Blazor will be as popular as ASP.NET web forms.

**Comment by AshkanAfsharpour** [comment]
Hey @jatinshah404! I totally get what you're saying about Blazor's current situation in the market. The competition with JavaScript frameworks, especially in areas like admin dashboards, UI libraries, and templates, is definitely real. I’ve been working with .NET and Blazor, and I'd be down to collaborate on creating a UI kit project specifically for Blazor.

I think if we gather the best components from open-source Blazor libraries and focus on building out a solid, versatile admin template, we could really make a difference. We could even consider bringing in FluentUI, Radzen, and other community-driven libraries to give users a strong, free option comparable to some of the big JavaScript frameworks. Let me know if you’re interested, and we can start brainstorming!
